### PR TITLE
Extract lesson plan resource list and linkWithQueryParams

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Follow up to https://github.com/code-dot-org/code-dot-org/pull/39136#discussion_r586744057. 

Some cleanup to reduce code duplication across several components:
- Extract listing resources to its own component in order to deduplicate the formatting logic
- Extract `linkWithQueryParams` into `utils.js`

## Testing story

manual testing, added a couple small tests but mostly relying on existing tests

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
Thanks for following up!

File: apps/src/utils.js [hunk 1]
```diff
@@ -743,6 +743,15 @@ export function stringifyQueryParams(params) {
   return '?' + keys.map(key => `${key}=${params[key]}`).join('&');
 }
 
+/**
+ * Takes a link, looks for params already in the current URL
+ * and generates a new link with those params
+ */
+export function linkWithQueryParams(link) {
```
Comment (dmcavoy) [inline apps/src/utils.js:750]: I wonder if there are other places doing this same thing which we could update? Totally not blocking just figured I'd ask.

**Review by dmcavoy — APPROVED** [review]

File: apps/src/utils.js (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/utils.js:750]: oh good thought. Even a quick look shows me I missed one spot in our lesson overview code 🙃 I'll take a quick look to see if there are other places as well but I'll definitely update that spot

Comment (bethanyaconnor) [reply apps/src/utils.js:750]: Found a few more places in the dropdown and the editors

**Pull request merged**